### DISCUSSION TITLE
filter data by custom date

### DISCUSSION THREAD

**Discussion by apujayed** [pocketbase/pocketbase#4109] [Q&A] [answered]
There was an issue that if i try to filter data between two date like this 
`created >="2023-01-07" && created <="2023-02-07"`
its working but when trying to filter only one date  with this same code
`created >="2023-01-07" && created <="2023-01-07"`
than no data found how to solve this issue

**Comment by ganigeorgiev** [comment]
If you wan to filter by day range you need to provide the full datetime string ranges, aka. the start and end time parts. Please see https://github.com/pocketbase/pocketbase/discussions/2833#discussioncomment-6336756.

Also note that if you use the JS SDK and the [`pb.filter()` helper](https://github.com/pocketbase/js-sdk?tab=readme-ov-file#binding-filter-parameters) then you can assign plain JS `Date` object as placeholder param value and the SDK will take care to convert it to the appropriate date format.

**Comment by apujayed** [comment]
![Screenshot (310)](https://github.com/pocketbase/pocketbase/assets/95185551/fd4962e9-2456-4a13-b6aa-d18fe29a4e7b)
based on the custom date field if i want to filter data between 2023-01-07 and 2024-01-08 another scenerio if i want to filter 2024-01-08 to 2024-01-08 than whats the condition i need to write inside filter:''?

`custom_date>="2024-01-08 00:00:00.000Z" && custom_date<="2024-01-08 00:00:00.000Z"`

this code not work

**Reply by ganigeorgiev** [reply, marked as answer]
As mentioned earlier if you need a day range you'll have to specify the time parts of your day(s) range, aka. the complete `date` field string format `Y-m-d H:i:s.uZ`. For example:

```js
created >= '2024-01-07 00:00:00.000Z' && created <= "2024-01-08 23:59:59.999Z"
```
The above should return all records with "created" date between the beginning of `2024-01-07` and the end of `2024-01-08`.

**Reply by ganigeorgiev** [reply]
You edited your last part of the comment but to avoid confusion if someone ended up here, your example doesn't work because you are targeting in both constraints **only** the start of the 2024-01-08 day. Again, if you want a day range you need to specify the start and end times.

**Reply by apujayed** [reply]
I got your point. It works. Thank you.

**Comment by kabapy** [comment]
remember not to use the standard ISO time stamp as it will give you issues while filtering in pocketbase

updated<'2024-11-03T12:46:37.849Z'  will not work .. giving wrong filter results
updated<'2024-11-03 12:46:37.849Z'  Will work !! (remove the T)

this is unexpected bug haunted me for months
